File: scripts/inspect-db.js
const Database = require('better-sqlite3');
const path = require('path');

function inspect(dbPath, table) {
  console.log('Inspecting:', dbPath);
  try {
    const db = new Database(dbPath, { readonly: true });
    const cols = db.prepare(`PRAGMA table_info('${table}')`).all();
    console.log(JSON.stringify(cols, null, 2));
    db.close();
  } catch (err) {
    console.error('Failed to read DB:', dbPath, err.message);
  }
}

const runtimeDb = path.join(process.env.APPDATA || '', 'Gestion des Arrivages', 'data', 'dev.db');
const workspaceDb = path.join(__dirname, '..', 'prisma', 'dev.db');

inspect(runtimeDb, 'PriceEntry');
inspect(workspaceDb, 'PriceEntry');

File: scripts/inspect_db.py
import sqlite3, json, os
DB='prisma/dev.db'
if not os.path.exists(DB):
    print('DB not found:', DB)
    raise SystemExit(2)
conn=sqlite3.connect(DB)
cur=conn.execute("PRAGMA table_info('ProductCompatibility')")
rows=cur.fetchall()
cols=[{"cid":r[0],"name":r[1],"type":r[2],"notnull":r[3],"dflt_value":r[4],"pk":r[5]} for r in rows]
print(json.dumps(cols, indent=2, ensure_ascii=False))
conn.close()
